src/build/__init__.py

```diff
@@ -151,10 +154,23 @@ def __init__(self, srcdir='.', config_settings=None):  # type: (str, Optional[Co
 
         self._backend = self._build_system['build-backend']
 
-        self.hook = pep517.wrappers.Pep517HookCaller(
-            self.srcdir, self._backend, backend_path=self._build_system.get('backend-path')
+        self._hook = pep517.wrappers.Pep517HookCaller(
+            self.srcdir,
+            self._backend,
+            backend_path=self._build_system.get('backend-path'),
+            python_executable=python_executable,
         )
 
+    @property
+    def python_executable(self):  # type: () -> unicode
```

Comment:
Any reason you're not using here ``typing.Text``, and also annotate as such the python executable in the constructor to not have this weird isinstance check?